components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/GitpodClientProjectSessionTracker.kt

```diff
@@ -35,6 +58,140 @@ class GitpodClientProjectSessionTracker(
         }
     }
 
+    private fun isExposedServedPort(port: Status.PortsStatus?) : Boolean {
+        if (port === null) {
+            return false
+        }
+        return port.served && port.hasExposed()
+    }
+
+    private fun showOpenServiceNotification(port: PortsStatus, offerMakePublic: Boolean = false) {
+        val message = "A service is available on port ${port.localPort}"
+        val notification = manager.notificationGroup.createNotification(message, NotificationType.INFORMATION)
+
+        val openBrowserAction = NotificationAction.createSimple("Open Browser") {
+            openBrowser(port.exposed.url)
+        }
+        notification.addAction(openBrowserAction)
+
+        if (offerMakePublic) {
+            val makePublicLambda = {
+                runBlocking {
+                    makePortPublic(info.workspaceId, port)
+                }
+            }
+            val makePublicAction = NotificationAction.createSimple("Make Public", makePublicLambda)
+            notification.addAction(makePublicAction)
+        }
+
+        ClientId.withClientId(session.clientId) {
+            val project = RestService.getLastFocusedOrOpenedProject()
```

Comment:
you need to find a project associated with session, see how it is done in cli

Reply:
Let's do a step back, in [GitpodManager.kt#L175](https://github.com/gitpod-io/gitpod/blob/main/components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/GitpodManager.kt#L175), the function `notification.notify(null)` receives `null` as `projectId`. 

Initially, I copied that approach, but given that we have the project, I thought it would be better to pass it, so that each session will receive the notification properly. I think it must be working anyway with `null` because we only use 1 session, perhaps.

Anyway, the retrieve of the project, I got it from [GitpodCLIService.kt](https://github.com/gitpod-io/gitpod/blob/main/components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/GitpodCLIService.kt#L72).

So, I am not sure what to change. 

1. Shall we keep `null`?
2. If not, can you point me to where I can get a project associated with session? I assumed `getLastFocusedOrOpenedProject` would do that 🤔

Reply:
does `session.project` work? 

if not let's keep `null`

Reply:
`session.project` seems to work! Thanks 🙏